clearpath_hardware_interfaces/src/a300/fan_control.cpp

```diff
@@ -0,0 +1,182 @@
+#include "clearpath_hardware_interfaces/a300/fan_control.hpp"
+
+std::string a300_cooling::thermalStatusToString(a300_cooling::ThermalStatus status)
+{
+  switch (status)
+  {
+    case a300_cooling::ThermalStatus::Normal:
+      return "Normal";
+    case a300_cooling::ThermalStatus::LowError:
+      return "LowError";
+    case a300_cooling::ThermalStatus::LowWarning:
+      return "LowWarning";
+    case a300_cooling::ThermalStatus::HighWarning:
+      return "HighWarning";
+    case a300_cooling::ThermalStatus::HighError:
+      return "HighError";
+    default:
+      return "Unknown";
+  }
+}
+
+a300_cooling::ThermalSensor::ThermalSensor(float low_error, float low_warning, float high_warning, float high_error)
+  : reading_(INITIAL_READING), low_error_(low_error), low_warning_(low_warning), high_warning_(high_warning), high_error_(high_error) {}
+
+a300_cooling::ThermalSensor::ThermalSensor()
+  : reading_(INITIAL_READING), low_error_(0.0), low_warning_(0.0), high_warning_(0.0), high_error_(0.0) {}
+
+void a300_cooling::ThermalSensor::setValue(float value) { this->reading_ = value; }
+
+a300_cooling::ThermalStatus a300_cooling::ThermalSensor::getStatus() const
+{
+  if (this->reading_ < this->low_error_) return a300_cooling::ThermalStatus::LowError;
+  if (this->reading_ < this->low_warning_) return a300_cooling::ThermalStatus::LowWarning;
+  if (this->reading_ > this->high_error_) return a300_cooling::ThermalStatus::HighError;
+  if (this->reading_ > this->high_warning_) return a300_cooling::ThermalStatus::HighWarning;
+  return a300_cooling::ThermalStatus::Normal;
+}
+
+a300_cooling::ThermalSensors::ThermalSensors()
+{
+  sensors =
+  {
+    {"mcu",          ThermalSensor{-20, 0, 60, 80}},
+    {"fan1",         ThermalSensor{-20, 0, 60, 80}},
+    {"fan2",         ThermalSensor{-20, 0, 60, 80}},
+    {"fan3",         ThermalSensor{-20, 0, 60, 80}},
+    {"fan4",         ThermalSensor{-20, 0, 60, 80}},
+    {"5v_inductor",  ThermalSensor{-20, 0, 60, 80}},
+    {"main_gnd_lug", ThermalSensor{-20, 0, 60, 80}},
+    {"dcdc_24v",     ThermalSensor{-20, 0, 60, 80}},
+    {"dcdc_12v",     ThermalSensor{-20, 0, 60, 80}},
+    {"battery",      ThermalSensor{-20, 0, 60, 80}},
+    {"pcb_motor1",   ThermalSensor{-20, 0, 60, 80}},
+    {"mcu_motor1",   ThermalSensor{-20, 0, 60, 80}},
+    {"pcb_motor2",   ThermalSensor{-20, 0, 60, 80}},
+    {"mcu_motor2",   ThermalSensor{-20, 0, 60, 80}},
+    {"pcb_motor3",   ThermalSensor{-20, 0, 60, 80}},
+    {"mcu_motor3",   ThermalSensor{-20, 0, 60, 80}},
+    {"pcb_motor4",   ThermalSensor{-20, 0, 60, 80}},
+    {"mcu_motor4",   ThermalSensor{-20, 0, 60, 80}},
```

Comment:
I think some of these had different thresholds, right?

Reply:
Just the battery was different and it was updated.